File: 17_03_2024_Count_Pairs_whose_sum_is_equal_to_X.py
class Solution:
    def countPair(self, head1, head2, n1, n2, x):
        '''
        head1:  head of linkedList 1
        head2:  head of linkedList 2
        n1:  len of  linkedList 1
        n2:  len of linkedList 1
        x:   given sum
        '''
        D = dict()
        count = 0
        
        while head1:
            D[x - head1.data] = D.get(x-head1.data, 0)+1
            head1 = head1.next
            
        while head2:
            count += D.get(head2.data, 0)
            head2 = head2.next
        return count

File: 17-03-2024_Count_Pairs_whose_sum_is_equal_to_X.cpp
class Solution{
  public:
    // your task is to complete this function
    int countPairs(struct Node* head1, struct Node* head2, int x) {
        // Code here
          unordered_map<int,int> mp;
        int count = 0;
        while(head1 != NULL){
            mp[x-head1->data]++;
            head1 = head1->next;
        }
        
        while(head2 != NULL){
            if(mp[head2->data] >= 1){
                count++;
            }
            head2 = head2->next;
        }
        return count;
    }
};
